### PR TITLE
(Clean-up) balances and allowances display 

### PR DESCRIPTION
**Pull request by isreallee82**
**Before submitting this PR, please make sure**:

- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

Addittion of a new gateway balance command that reports balance and allowance  for single connected wallets.
Adds allowance to balance display
Reafctor of all the necessary function to make the command work.
Fix other related bugs.

Tests performed by the developer:

Tips for QA testing:

### CODE REVIEW

**Comment by nikspz** [comment]
Initial tests:

Cloned and installed PR6787 and gateway latest development

1) gateway balance showing 0 balance for dexalot (when other 4 exchanges connected as well), however dexalot balance showing when using `gateway balance dexalot_avalanche_dexalot` command

gateway balance showing 0 balance for dexalot

image

2) Allowances for uniswap_ethereum_goerli and uniswap_ethereum_arbitrum showing huge numbers, seems expected on allowances showed the same

image


image
image

**Comment by isreallee82** [comment]
fixed the dexalot error display to return 0 value also on single exchange request if not connected, As for the allowances with huge amount, I am sure that was expected since the bot is designed to approve them @nikspz

**Comment by nikspz** [comment]
@isreallee82 currently the same issue for dexalot

Steps: 
1. Clone and install PR6787 commit 1abea4a and gateway latest development
2. gateway connect dexalot_avalanche
3. check gateway balance or gateway balance dexalot

Actual:
showing 0 balance for dexalot


gateway balance showing 0 balance for dexalot

**Comment by isreallee82** [comment]
sorry for late response @nikspz  As mentioned earlier, I've configured the display to return a default value of 0 for any exchange that doesnt use allowance since dexalot is a club spot, as evidenced in the gateway request terminal. also configured to match the index since balances and allowances are pulled from exchanges on the gateway conference file

File: hummingbot/client/command/gateway_command.py [hunk 1]
```diff
@@ -445,51 +442,167 @@ async def _prompt_for_wallet_address(
...
+                    # Handle errors in allowance_responses_list
+                    if allowance_resp.get("approvals") is None:
+                        allowance = Decimal("0")
+                    else:
+                        allowance = allowance_resp["approvals"].get(token, Decimal("0"))
+
+                    rows.append({
+                        "Symbol": token.upper(),
+                        "Balance": PerformanceMetrics.smart_round(Decimal(str(bal)), 4),
+                        "Allowance": allowance,
```
Comment (fengtality) [inline hummingbot/client/command/gateway_command.py:498]: For display purposes, I suggest showing "999999+" if `allowance > 999999`. This should address @nikspz's comment.

**Review by fengtality — CHANGES_REQUESTED** [review]

**Comment by isreallee82** [comment]
noted @nikspz @fengtality I will refactor, test and push asap.

**Comment by nikspz** [comment]
@isreallee82 

commit 1b8bf44 issue for dexalot

Steps:
Clone and install PR6787 commit 1b8bf44 and gateway latest development
gateway connect dexalot_avalanche
check gateway balance or gateway balance dexalot

Actual:
showing 0 balance for dexalot


image

Crosschecked, showed on Client+gateway development, but broken on this PR

**Comment by isreallee82** [comment]
should work now as expected @nikspz

**Comment by fengtality** [comment]
@isreallee82 Sorry for the delay in reviewing this PR. Everything looks good except for this issue I found where the connector filter doesn't seem to work. In the examples below, it only displays the first item rather than the selected conenctor.
Screenshot 2024-03-08 at 3 54 17 PM

**Comment by isreallee82** [comment]
Ohh it’s fine @fengtality i will look into that and make a push asap

**Comment by isreallee82** [comment]
fix that @fengtality

**Review by nikspz — APPROVED** [review]
Test performed:

- Re-check connectors based `gateway balance` with allowances ✔
- Review gateway balance now shows balances for separate connectors and shows correct walelts ✔
- command connector-tokens works (added tokens to balance)✔
- Review allowances showing 999+ for big numbers now ✔
- Created/started amm_arb strategy succesfully for tokens with allowances ✔

Note: Review delay for checking single gateway balance could up to ~7-8sec 📈
Note: Review gateway balance for too many connectors returns Timeout Error ❌(reproduced on development earlier)



Checking gateway balance for single connector with allowances takes ~7-8 seconds: 
checking balance with allowances is too slow

Checking gateway balance for multiple connectors takes time (if too many connectors command will crash with Timeout Error)

multiple connectors checking balance with allowances is too slow

**Review by fengtality — APPROVED** [review]
LGTM

**Pull request merged**